src/Schema/SchemaInterface.php

```diff
@@ -0,0 +1,489 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Schema;
+
+use Yiisoft\Db\Cache\SchemaCache;
+use Yiisoft\Db\Connection\ConnectionInterface;
+use Yiisoft\Db\Constraint\CheckConstraint;
+use Yiisoft\Db\Constraint\Constraint;
+use Yiisoft\Db\Constraint\DefaultValueConstraint;
+use Yiisoft\Db\Constraint\ForeignKeyConstraint;
+use Yiisoft\Db\Constraint\IndexConstraint;
+use Yiisoft\Db\Exception\Exception;
+use Yiisoft\Db\Exception\InvalidCallException;
+use Yiisoft\Db\Exception\InvalidConfigException;
+use Yiisoft\Db\Exception\NotSupportedException;
+use Yiisoft\Db\Query\QueryBuilder;
+
+interface SchemaInterface
+{
+    public const SCHEMA = 'schema';
+    public const PRIMARY_KEY = 'primaryKey';
+    public const INDEXES = 'indexes';
+    public const CHECKS = 'checks';
+    public const FOREIGN_KEYS = 'foreignKeys';
+    public const DEFAULT_VALUES = 'defaultValues';
+    public const UNIQUES = 'uniques';
+
+    public const TYPE_PK = 'pk';
+    public const TYPE_UPK = 'upk';
+    public const TYPE_BIGPK = 'bigpk';
+    public const TYPE_UBIGPK = 'ubigpk';
+    public const TYPE_CHAR = 'char';
+    public const TYPE_STRING = 'string';
+    public const TYPE_TEXT = 'text';
+    public const TYPE_TINYINT = 'tinyint';
+    public const TYPE_SMALLINT = 'smallint';
+    public const TYPE_INTEGER = 'integer';
+    public const TYPE_BIGINT = 'bigint';
+    public const TYPE_FLOAT = 'float';
+    public const TYPE_DOUBLE = 'double';
+    public const TYPE_DECIMAL = 'decimal';
+    public const TYPE_DATETIME = 'datetime';
+    public const TYPE_TIMESTAMP = 'timestamp';
+    public const TYPE_TIME = 'time';
+    public const TYPE_DATE = 'date';
+    public const TYPE_BINARY = 'binary';
+    public const TYPE_BOOLEAN = 'boolean';
+    public const TYPE_MONEY = 'money';
+    public const TYPE_JSON = 'json';
+
+    public function createQueryBuilder(): QueryBuilder;
+
+    /**
+     * @return QueryBuilder the query builder for this connection.
+     */
+    public function getQueryBuilder(): QueryBuilder;
```

Comment:
Do we need separate `createQueryBuilder()` and `getQueryBuilder()`?

Reply:
it's be fixed in next PR